### PR TITLE
Add config option for continue commenting

### PR DESCRIPTION
**Pull request by TornaxO7**
This fixes https://github.com/helix-editor/helix/issues/1730#issuecomment-2497839562

@johnbchron may I ask if you could try my PR out? My PR adds the field in the `[editor]` section. Just add a `continue-comment = false` line and you should be good to go.

I'm also wondering if it might be worth it to add a command for this: `:set-continue-comment [true/false]`

### CODE REVIEW

**Review by pascalkuthe — APPROVED** [review]
don't love adding new  config options but this feature has a pretty big effect and I can see not everyone liking it I suppose

**Comment by johnbchron** [comment]
@TornaxO7 works for me.

@pascalkuthe I agree that it's a reasonable default but in some projects it adds a bunch of unnecessary work to have permanently enabled. I appreciate you allowing the config option.

File: book/src/editor.md [hunk 1]
```diff
@@ -31,6 +31,7 @@
 | `line-number` | Line number display: `absolute` simply shows each line's number, while `relative` shows the distance from the current line. When unfocused or in insert mode, `relative` will still show absolute line numbers | `absolute` |
 | `cursorline` | Highlight all lines with a cursor | `false` |
 | `cursorcolumn` | Highlight all columns with a cursor | `false` |
+| `continue-comment` | if helix should automatically add a (line-)comment token, if you create a new line inside a comment. | `false` |
```
Comment (the-mikedavis) [inline book/src/editor.md:34]:
```suggestion
| `continue-comments` | if helix should automatically add a line comment token if you create a new line inside a comment. | `false` |
```

I'm not sure we can accurately continue block comments but if we do add that we can just edit the text here. I would also call this `continue-comments` instead and there's a stray `,` that can be removed here

Comment (the-mikedavis) [reply book/src/editor.md]: The option and comment in the code should be renamed to match this too

Comment (TornaxO7) [reply book/src/editor.md]: Fair point. Changes applied.

**Review by the-mikedavis — APPROVED** [review]

**Review by the-mikedavis — APPROVED** [review]

File: book/src/editor.md [hunk 2]
```diff
@@ -31,6 +31,7 @@
 | `line-number` | Line number display: `absolute` simply shows each line's number, while `relative` shows the distance from the current line. When unfocused or in insert mode, `relative` will still show absolute line numbers | `absolute` |
 | `cursorline` | Highlight all lines with a cursor | `false` |
 | `cursorcolumn` | Highlight all columns with a cursor | `false` |
+| `continue-comments` | if helix should automatically add a line comment token if you create a new line inside a comment. | `false` |
```
Comment (jlebon) [inline book/src/editor.md:34]: Hmm, typo here I think re. the default: s/false/true/

Comment (TornaxO7) [reply book/src/editor.md:34]: Ah oof, true ._.

**Comment by llakala** [comment]
I'd love if there was also a command for the old behavior, so it could be bindable, rather than simply toggleable in general. Is there an existing workaround for this, or should I make an issue?

**Pull request merged**